src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -2162,8 +2178,20 @@ public Status publishError(Status status) {
     return status;
   }
 
+  /**
+   * All status are published through here error, warn and info.
+   * Usually these are events which are sent back to the user. 
+   * They are now available on th statusList queue with a buffer size of 20
+   */
   @Override
   public Status publishStatus(Status status) {
+    statusList.add(status);
+    int size = statusList.size();
+    if (size > 20) {
```

Comment:
All of this could be replaced with a call to offer(), and if false just call poll once. Would need a lock around the whole section, can use a read write lock instead of synchronized so that multiple threads can read at the same time so long as none are writing.

Would need to change the status list to a queue that has a fixed capacity to use offer like this